### ISSUE TITLE
Add support for `ws` and `wss`.

### ISSUE THREAD

**Issue by birdsarah** [mozilla/domain_utils#14]
We should add support for `ws` and `wss`. (https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_client_applications)

It would be helpful to be able to retrieve eTLD+1 from these as well.

_Originally posted by @englehardt in https://github.com/mozilla/domain_utils/pull/12_

**Comment by birdsarah** [comment]
Suggest a new flag `parse_ws_urls` which parses websocket scheme urls `ws` and `wss`.

We could also revamp and have an option for user to pass in schemes to be parsed, at their own risk.

So the first option would look like this:

```python
def get_stripped_url(url, parse_ws_urls=True, scheme=False, drop_non_http=False, use_netloc=True, extractor=None):

    ....
    # Handle non http schemes
    schemes_to_parse =  ['http', 'https', '']
    if parse_ws_urls is True:
        schemes_to_parse += ['ws', 'wss']
    if _scheme not in schemes_to_parse:
        if drop_non_http is True:
            return ''
        else:
            return url
     ....
```

Option 2

```python
DEFAULT_SCHEMES = ['http', 'https', 'ws', 'wss']
def get_stripped_url(url, schemes_to_parse=DEFAULT_SCHEMES, scheme=False, drop_non_http=False, use_netloc=True, extractor=None):
    ....
    # Handle non http schemes
    if _scheme not in schemes_to_parse + ['']:
        if drop_non_http is True:
            return ''
        else:
            return url
     ....
```

As I'm writing this out a couple of things are occurring to me with this API.

I think the following collection of flags may make more sense and be more flexible:
* `schemes_to_parse` - list of schemes for which we want to try and do the stemming process (removing scheme and params (and maybe path #10))
* `no_scheme_as_http` - where a url with an empty scheme is prepended with `http://` (there currently isn't a feature flag for this)
* `return_unparsed` - replacement for `drop_non_http` to be more generic.

cc @englehardt for input?

**Comment by englehardt** [comment]
> As I'm writing this out a couple of things are occurring to me with this API.
> 
> I think the following collection of flags may make more sense and be more flexible:
> 
>     * `schemes_to_parse` - list of schemes for which we want to try and do the stemming process (removing scheme and params (and maybe path #10))

I'm in favor of specifying ws and wss explicitly (and adding tests). I think that's safer than allowing users to specify random schemes that might parse in unexpected ways.

I don't think we should drop the `scheme` and `path` options. sometimes you want the scheme and sometimes not. I suspect the same is true for path.

> 
>     * `no_scheme_as_http` - where a url with an empty scheme is prepended with `http://` (there currently isn't a feature flag for this)

Makes sense. It could also be `default_to_http`? (I'm fine with either).

> 
>     * `return_unparsed` - replacement for `drop_non_http` to be more generic.

Sounds good!

**Comment by birdsarah** [comment]
> I don't think we should drop the `scheme` and `path` options. sometimes you want the scheme and sometimes not. I suspect the same is true for path.

Oh yes, I agree. Sorry that wasn't clear from above. (the note in parens was a definition of stemming not a proposal).

New signature proposal:

```python
HTTP = 'http'
DEFAULT_SCHEMES = [HTTP, 'https', 'ws', 'wss']
def get_stripped_url(
    url, 
    schemes_to_parse=DEFAULT_SCHEMES,  # I think you were saying you don't like this
    return_unparsed=True, 
    scheme_default=HTTP,  # This param will be passed into urlparse (see #18) 
    scheme=False, 
    path=True,  # To be implemented in #10
    use_netloc=True, 
    extractor=None):
```

**Comment by englehardt** [comment]
I see, that makes sense! Thanks for clarifying.

It still feels safer to only allow parsing of things that we know will parse correctly to avoid any foot-gun scenarios. But if you see a compelling reason to give users the option to parse more then I won't block it.

**Comment by birdsarah** [comment]
I'm a little torn. There's some investigating I want to do right now for example, that's just looking at webpack urls. But, I totally see your point about footguns. My main concern was an explosion of flags for handling various kinds of schemes which the `ws` request has highlighted.

I thought about a middle ground be using the proposed "DEFAULT_SCHEMES" above but then checking the contents of DEFAULT_SCHEMES and it contains an entry not in the defaults list `http`, `https`, `ws`, `wss` that we raise a warning. My concern about this warning idea is that warnings and stuff can get eaten in distributed systems like spark.

I am fine with the following implementation and then seeing how annoying it is that i'll need to roll my own for a non standard case i.e. is my current investigation a one-off or something that it would be good if this library handled.

```python
HTTP = 'http'
HTTPS = 'https'
WS = 'ws'
WSS = 'wss'
def get_stripped_url(
    url, 
    return_unparsed=True, 
    scheme_default=HTTP,  # This param will be passed into urlparse (see #18) 
    parse_ws=True,
    scheme=False, 
    path=True,  # To be implemented in #10
    use_netloc=True, 
    extractor=None):
    ....
    # Handle non http schemes
    schemes_to_parse = [HTTP, HTTPS]
    if parse_ws is True:
        schemes_to_parse += [WS, WSS]
    if _scheme not in schemes_to_parse:
        if return_unparse is True:
            return url
        else:
            return ''
```
